Sources/FoundationEssentials/FileManager/FileManager+Files.swift

```diff
@@ -503,9 +603,47 @@ extension _FileManagerImpl {
             #endif
             return attributes
         }
+#endif
     }
     
     func attributesOfFileSystem(forPath path: String) throws -> [FileAttributeKey : Any] {
+#if os(Windows)
+        return try path.withNTPathRepresentation { pwszPath in
+            let dwLength: DWORD = GetFullPathNameW(pwszPath, 0, nil, nil)
+            guard dwLength > 0 else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+
+            var szVolumeName: [WCHAR] = Array<WCHAR>(repeating: 0, count: Int(dwLength + 1))
```

Comment:
`withUnsafeTemporaryAllocation` might be a better choice here than an array - also do we need the `+1`?

Reply:
Aye and nay.